skglm/experimental/sqrt_lasso.py

```diff
@@ -209,7 +222,10 @@ def path(self, X, y, alphas=None, eps=1e-3, n_alphas=10):
                 # save coef despite not converging
                 # coef_init holds a ref to coef
                 coef = coef_init
-                res_norm = norm(y - X @ coef)
+                if self.fit_intercept:
+                    res_norm = norm(y - X @ coef[:-1] - coef[-1])
+                else:
+                    res_norm = norm(y - X @ coef)
```

Comment:
WDYT about this implementation to make it more precise?

```suggestion
                X_coef = X @ coef if not fit_intercept else X @ coef[:-1] + coef[-1]
                res_norm = norm(y - X_coef)
```